tests/dragonfly/acl_family_test.py

```diff
@@ -59,21 +59,133 @@ async def test_acl_setuser(async_client):
+
+    # This should fail, vlad does not have @sortedset
+    with pytest.raises(redis.exceptions.ResponseError):
+        await async_client.execute_command("ZADD myset 1 two")
+
+    result = await async_client.execute_command("AUTH default nopass")
+    assert result == "OK"
+
+    # Make vlad an admin
+    await async_client.execute_command("ACL SETUSER vlad -@string")
+    assert result == "OK"
+
+    result = await async_client.execute_command("AUTH vlad mypass")
+    assert result == "OK"
+
+    with pytest.raises(redis.exceptions.ResponseError):
+        await async_client.execute_command("GET foo")
+
+    result = await async_client.execute_command("AUTH default nopass")
+    assert result == "OK"
+
+    # Vlad goes rogue starts giving admin stats to random users
+    await async_client.execute_command("ACL SETUSER adi >adi +@admin")
+    assert result == "OK"
+
+    # Vlad can now execute everything
+    await async_client.execute_command("ACL SETUSER vlad +@all")
+    assert result == "OK"
+
+    await async_client.execute_command("ZADD myset 1 two")
+    assert result == "OK"
+
+
+@pytest.mark.asyncio
+async def test_acl_categories_multi_exec_squash(df_local_factory):
+    df = df_local_factory.create(multi_exec_squash=True, port=1111)
+
+    df.start()
+
+    client = aioredis.Redis(port=df.port)
+    res = await client.execute_command("ACL SETUSER kk ON >kk +@transaction +@string")
+    assert res == b"OK"
+
+    res = await client.execute_command("AUTH kk kk")
+    assert res == b"OK"
+
+    await client.execute_command("MULTI")
+    assert res == b"OK"
+    for x in range(33):
+        await client.execute_command(f"SET x{x} {x}")
+    await client.execute_command("EXEC")
+
+    client = aioredis.Redis(port=df.port)
+    await client.close()
+
+    # NOPERM while executing multi
+    await client.execute_command("ACL SETUSER kk -@string")
+    assert res == b"OK"
+    await client.execute_command("AUTH kk kk")
+    assert res == b"OK"
+    await client.execute_command("MULTI")
+    assert res == b"OK"
+
+    with pytest.raises(redis.exceptions.ResponseError):
+        await client.execute_command(f"SET x{x} {x}")
+    await client.close()
+
+    # NOPERM between multi and exec
+    admin_client = aioredis.Redis(port=df.port)
+    res = await client.execute_command("ACL SETUSER kk +@string")
+    assert res == b"OK"
+
+    client = aioredis.Redis(port=df.port)
+    res = await client.execute_command("AUTH kk kk")
+    assert res == b"OK"
+    # CLIENT has permissions, starts MULTI and issues a bunch of SET commands
+    await client.execute_command("MULTI")
+    assert res == b"OK"
+    for x in range(33):
+        await client.execute_command(f"SET x{x} {x}")
+
+    # ADMIN revokes permissions
+    res = await admin_client.execute_command("ACL SETUSER kk -@string")
+    assert res == b"OK"
+
+    res = await client.execute_command("EXEC")
+    # TODO(we need to fix this, basiscally SQUASHED/MULTI transaction commands
```

Comment:
@dranikpg This little code needs some polishing. I will address this in a separate PR. Basically, we need to squash the error messages into one....